src/Core/mcts/UCT.cpp

```diff
@@ -1,6 +1,57 @@
 #include <UCT.h>
 
-UCT::UCT(EnvironmentBase &environment, boost::function<int(State)> default_policy)
-    : MCTSBase(environment, default_policy){};
+UCT::UCT(EnvironmentBase &environment) : MCTSInterface(environment), generator(std::mt19937(time(nullptr))){};
 
-void UCT::run(int n_searches) {}
+std::shared_ptr<SearchNode> UCT::m_tree_policy(std::shared_ptr<SearchNode> node) { return m_tpolicy.treePolicy(node); };
+
+Reward UCT::m_default_policy(State &state) { return m_defaultPolicy.defaultPolicy(state); };
+
+std::shared_ptr<SearchNode> UCT::m_best_child(std::shared_ptr<SearchNode> node, double c) {
+    int best_child = 0;
+    double best_score_so_far = -1;
```

Comment:
Depending on the domain, `-1` may not be a suitable score.
Use `std::numeric_limits<double>::lowest()`

Reply:
Should be in `#include <limits>`, may need to google